Assets/00_Content/Scripts/Taverns/Tavern.cs

```diff
@@ -8,37 +8,37 @@ public class Tavern : SingletonBehaviour<Tavern> {
 		[SerializeField] private int maxSittingGuests;
 		[SerializeField] private float maxHealth = 100;
 		[SerializeField] private float startingHealth = 100;
-		[SerializeField] private float startingMoney = 50;
-		[SerializeField] private float maxMoney = 100;
+		[SerializeField] private int startingMoney = 0;
 
-		private bool IsBankrupt => Money <= 0;
 		private bool IsDestroyed => Health <= 0;
 
 		private float health;
-		private float money;
+		private int money;
 
-		public event Action OnBankrupcy;
 		public event Action OnDestroyed;
 		public event Action OnMoneyChanges;
 		public event Action OnHealthChanges;
 
 		public float MaxHealth => maxHealth;
 		public float StartingHealth => startingHealth;
-		public float StartingMoney => startingMoney;
-		public float MaxMoney => maxMoney;
+		public int StartingMoney => startingMoney;
 
 		public float Health {
 			get => health;
 			set { health = Mathf.Round(Mathf.Clamp(value, 0, maxHealth)); }
 		}
 
-		public float Money {
+		public int Money {
 			get => money;
 			set {
-				float newMoney = Mathf.Round(Mathf.Clamp(value, 0, maxMoney));
+				int newMoney = Mathf.RoundToInt(value);
+				
 				if (money == newMoney)
 					return;
 
+				if (newMoney < 0)
+					newMoney = 0;
```

Comment:
This being done after the `if (money == newMoney)` check might cause the `OnMoneyChanges` event to be invoked even though the money did not change.